.circleci/config.yml

```diff
@@ -75,106 +66,327 @@ commands:
+          working_directory: ~/repo/ddp-workspace
+
+  setup-shared-env:
+    description: "Setup shared ENV vars used by multiple scripts"
+    parameters:
+      study_key:
+        type: string
+    steps:
+      - run:
+          name: Set environment variables to build <<parameters.study_key>>
+          command: |
+            echo 'export ANGULAR_PROJECT_NAME=ddp-<<parameters.study_key>>' >> $BASH_ENV
+            echo 'export ANGULAR_PROJECT_DIR_PATH=projects/ddp-<<parameters.study_key>>' >> $BASH_ENV
+            source $BASH_ENV
+            OUTPUT_DIR="${ANGULAR_PROJECT_DIR_PATH}/dist"
+            echo "export OUTPUT_DIR=$OUTPUT_DIR" >> $BASH_ENV
+            echo "export SHORT_GIT_SHA=`echo '<< pipeline.git.revision >>' | cut -c 1-7`" >> $BASH_ENV
+            echo 'export BUILDS_BUCKET=ddp-angular-builds' >> $BASH_ENV
+            source $BASH_ENV
+
+  setup-build-workspace:
+    description: "Setup workspace used by all projects"
+    steps:
+      - checkout:
+          path: ~/repo
       - restore_cache:
           key: *npm_cache_key
       - run:
-          name: <<parameters.study_key>> npm install
+          name: npm install
           command: npm install
           working_directory: ~/repo/ddp-workspace
       - save_cache:
           key: *npm_cache_key
           paths:
             - ./node_modules
+
+  build-app:
+    description: "Build DDP study app"
+    steps:
       - run:
-          name: <<parameters.study_key>> ng build
-          command: ng build $ANGULAR_PROJECT_NAME $NG_BUILD_OPTS --base-href=/ --output-path=$ANGULAR_PROJECT_DIR_PATH/dist --verbose=true --progress=true
+          name: ng build
+          command: |
+            set -u
+            ng build $ANGULAR_PROJECT_NAME --prod --aot --base-href=/ --output-path=$OUTPUT_DIR --verbose=true --progress=true
           working_directory: ~/repo/ddp-workspace
+
+  store-app-build:
+    description: "Copy build to cloud storage"
+    steps:
       - run:
-          name: <<parameters.study_key>> setup gcloud context
+          name: Create build archive
           command: |
-            echo $GCLOUD_SERVICE_KEY | gcloud auth activate-service-account --key-file=-
-            # ran out of time, but demonstration of basic idea extracting data from Vault directly. Needs to be wrapped in nicer way!
-            #gcloud --quiet config set compute/zone `vault read  -format=json  secret/pepper/$DEPLOY_ENV/v1/conf | jq -r '.data.gcp.computeZone'`
-      - run:
-          name: <<parameters.study_key>> deploy to GAE
-          command: gcloud app deploy --version="${CIRCLE_SHA1}" --project "broad-ddp-${ENVIRONMENT}" "${ANGULAR_PROJECT_DIR_PATH}/app.yaml"
+            set -u
+            DATE=`date +%F`
+            TAR_NAME="${ANGULAR_PROJECT_NAME}_${DATE}_${SHORT_GIT_SHA}.tar.gz"
+            TAR_PATH="/tmp/${TAR_NAME}"
```

Comment:
Note the format chosen. Included date because I wanted to a have a clue on when the build occurred (outside of just the timestamp cuz that can get messed up with a copy).